spec/bolt/catalog_spec.rb

```diff
@@ -96,5 +161,9 @@
       expect(result['resources'].map { |r| r['type'] }).to include('Notify')
       expect(result['resources'].count).to eq(4)
     end
+
+    it 'resolves local references when deserializing Puppet types' do
+      expect { catalog.compile_catalog(reference_request) }.not_to raise_error(/undefined method `\[\]'/)
```

Comment:
This test doesn't seem to fail when I run it _without_ the changes.

Reply:
That's interesting... it's failing for me when run on master.

Reply:
Ah I was running against a local checkout of Puppet. Apparently it was hitting a different error instead, which technically meets this condition. Making this just `expect { ... }.not_to raise_error` might be less sensitive.

Reply:
Updated